CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -208,52 +207,48 @@ class CameraFragment : Fragment() {
             // level >= 24, so if you only target 24+ you can remove this statement
             if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
                 requireActivity().sendBroadcast(
-                        Intent(Camera.ACTION_NEW_PICTURE, Uri.fromFile(photoFile)))
+                        Intent("android.hardware.action.NEW_PICTURE", Uri.fromFile(photoFile)))
             }
 
             // If the folder selected is an external media directory, this is unnecessary
             // but otherwise other apps will not be able to access our images unless we
             // scan them using [MediaScannerConnection]
-            val mimeType = MimeTypeMap.getSingleton()
-                    .getMimeTypeFromExtension(photoFile.extension)
-            MediaScannerConnection.scanFile(
-                    context, arrayOf(photoFile.absolutePath), arrayOf(mimeType), null)
+            val mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(photoFile.extension)
+            if(!requireActivity().isFinishing) {
```

Comment:
This fixes a force-close, when pressing the back button too soon.

Reply:
I am unable to reproduce. Please create a separate issue with reproduction steps, and address this in a different PR.

Reply:
It is indeed difficult to reproduce, because one needs to press back at a certain point in time.

Reply:
That wouldn't make sense, because pressing the gallery button switches fragments while the activity should never go into finishing state. Please open a separate issue to track this down and let's handle it in a separate PR.